plugins/unicodelib/unicodelib.cpp

```diff
@@ -718,6 +719,43 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+UnicodeString excludeNthWord(RuleBasedBreakIterator& bi, unsigned textLen, UChar const * text, unsigned n)
+{
+    UnicodeString source(text, textLen);
+    bi.setText(source);
+    int32_t idx = bi.first();
+    int32_t wordidx = 0;
+    unsigned wordBeginning = 0;
+    while (idx != BreakIterator::DONE) {
+        int breakType = bi.getRuleStatus();
+        if (breakType != UBRK_WORD_NONE) {
+            // Exclude spaces, punctuation, and the like.
+            //   A status value UBRK_WORD_NONE indicates that the boundary does
+            //   not start a word or number.
+            if (++wordidx == n) {
+                if (n == 1) {
+                    wordBeginning = 0;
+                }
+                unsigned wordEnd;
+                do {
+                    wordEnd = idx;
+                    idx = bi.next();
+                } while (bi.getRuleStatus() == UBRK_WORD_NONE && idx != BreakIterator::DONE);
+                return source.removeBetween(wordBeginning, wordEnd);
+            }
+        }
+        wordBeginning = idx;
+        idx = bi.next();
+        if (idx == BreakIterator::DONE && wordidx < 1) {
```

Comment:
I think you can simplify the code by moving these two conditions outside the loop.